### PR TITLE
glibc: add UTF-8 support

### PR DESCRIPTION
**Pull request by thoradia**
LibreELEC does currently not support UTF-8, whereas OpenELEC does.
This pull request merely includes corresponding OpenELEC code, for discussion.
It has neither been tested, nor optimized (eg, current Generic ships gzip-ed charmaps).

### CODE REVIEW

File: packages/devel/glibc/package.mk [hunk 1]
```diff
@@ -134,6 +134,17 @@ post_makeinstall_target() {
 
 # remove locales and charmaps
   rm -rf $INSTALL/usr/share/i18n/charmaps
+  if [ -n "$GLIBC_LOCALES" ]; then
+    mkdir -p $INSTALL/usr/lib/locale
+    for locale in $GLIBC_LOCALES; do
+      echo ">>> install inputfile $(echo $locale | cut -f1 -d ".") with charmap $(echo $locale | cut -f2 -d ".") as $locale <<<"
+      I18NPATH=../localedata \
+      $ROOT/$TOOLCHAIN/bin/localedef \
```
Comment (escalade) [inline packages/devel/glibc/package.mk]: master does not use $ROOT anymore, also you need to add the eglibc-localedef:host package to get localedef in the toolchain. Most Linux distributions ship localedef by default though, so I guess you could remove the path entirely and assume that the host system has it.

**Comment by thoradia** [comment]
Fixed it with an [add-on](https://github.com/thoradia/LibreELEC.tv/commit/665ec97445538adb0980f62b322378e37136940b), called like this:
`LOCALE="$(thoradia-locale)"
export LANG="$(basename $LOCALE)"
export LOCPATH="$(dirname $LOCALE)"`

**Comment by chewitt** [comment]
@thoradia It would be great to adapt that add-on for our main add-on repo along with some python and systemd bits that allow users to configure the locale from the GUI. Possible?

**Comment by thoradia** [comment]
@chewitt 

en_US.UTF-8 is 1.6 MB
glibc charmaps is 17.0 MB (UTF-8 is 2.3 MB)
glibc locales is 6.2 MB

I will see what I can do.

At first thought:
1. add-on to generate one locale in storage/.config/locale
2. conditional binding in 98-busybox.conf: if custom locale /storage/.config/locale, then set LOCPATH and LANG accordingly

Would that fit your bill?

**Comment by chewitt** [comment]
Yeah, that's not going to happen when the average full OS image size is around 140MB - addon is the right direction. Locale content needs to be under /storage/.kodi/addons/<addon> so it's self contains and gets removed with the add-on if the user uninstalls it. Aside from that the idea sounds reasonable. I'm no developer but I'm sure the usual suspects will chime in with feedback once there's a submission. Thanks :)

**Pull request closed without merging**